functions/New-DbaDatabase.ps1

```diff
@@ -0,0 +1,317 @@
+It allows creation with multiple files, and sets all growth settings to be fixed size rather than percentage growth.
+
+.PARAMETER SqlInstance
+The SQL server instances to be connected to; can be passed in via the pipeline.
+
+.PARAMETER Force
+The force parameter will ignore some errors in the parameters and assume defaults.
+
+.PARAMETER WhatIf 
+Shows what would happen if the command were to run. No actions are actually performed. 
+
+.PARAMETER Confirm 
+Prompts you for confirmation before executing any changing operations within the command. 
+
+.PARAMETER Silent 
+Use this switch to disable any kind of verbose messages
+
+.NOTES 
+Original Author: Matthew Darwin (@evoDBA, naturalselectiondba.wordpress.com)
+Tags: Database
+	
+Website: https://dbatools.io
+Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+.LINK
+https://dbatools.io/New-DbaAgentJob
+
+.EXAMPLE   
+New-DbaAgentJob -SqlInstance sql1 -DatabaseName 'TestDatabase' -UserDataFileSize 128 -UserDataFileMaxSize 1024 -UserDataFileGrowth 128 `
+-LogSize 128 -LogGrowth 128
+Creates a database named TestDatabase on instance sql1 with a user filegroup with a single file of 128MB
+
+#>
+
+    [cmdletbinding(SupportsShouldProcess = $true, ConfirmImpact = "Low")]
+    Param 
+    ( 
+    # set variables for the database
+            [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		    [Alias("ServerInstance", "SqlServer")]
+		    [DbaInstanceParameter[]]$SqlInstance
+          , [PSCredential]$SqlCredential  
+          , [parameter(Mandatory = $true)]
+            [string]$DatabaseName                      
+          , [double]$NumberOfFilesInUserFilegroup = 1
+
+    # optional db variables
+          , [boolean]$UseDefaultFileLocations = $true
+          , [string]$NonDefaultFileLocation
+          , [string]$NonDefaultLogLocation
+          , [string]$Collation
+          , [string]$RecoveryModel
+          , [string]$DatabaseOwner = "sa"
+
+    #set the user data size, maxsize and growth
+          , [double]$UserDataFileSize 
+          , [double]$USerDataFileMaxSize
+          , [double]$UserDataFileGrowth #use 0 for no growth
+
+    #set the log size and growth
+          , [double]$LogSize
+          , [double]$LogGrowth
+
+    #set the primary file size in MB (will be converted to kb later)
+          , [double]$PrimaryFileSize = 10
+          , [double]$PrimaryFileGrowth = 10
+          , [double]$PrimaryFileMaxSize = 100
+          		
+    #Switches
+          , [switch]$Force
+		  , [switch]$Silent
+    )
+
+    begin
+    {
+        #Check file directories passed in if not using defaults
+        if ($UseDefaultFileLocations -eq $false -and ($NonDefaultFileLocation -eq $Null -or $NonDefaultLogLocation -eq $Null))
+            {
+                Stop-Function -Message "Non Default file locations selected, but are not supplied" -Category InvalidData -ErrorRecord $_ -Target $instance -Continue
```

Comment:
Slight error in using `Stop-Function`:
```PowerShell
# Replace this
Stop-Function -Message "Non Default file locations selected, but are not supplied" -Category InvalidData -ErrorRecord $_ -Target $instance -Continue

# With this
Stop-Function -Message "Non Default file locations selected, but are not supplied" -Category InvalidData -ErrorRecord $_
return
```
 - `Continue` should only be used in process within a loop, as it can otherwise kill the function or script calling it
 - `-Target` is to help with debugging, by containing the item currently being processed. You aren't processing an item yet (It's the begin block), so it shouldn't be there yet